src/library/scala/collection/immutable/Map.scala

```diff
@@ -335,7 +333,16 @@ object Map extends MapFactory[Map] {
     }
   }
 
-  final class Map4[K, +V](key1: K, value1: V, key2: K, value2: V, key3: K, value3: V, key4: K, value4: V) extends AbstractMap[K, V] with StrictOptimizedIterableOps[(K, V), Iterable, Map[K, V]] {
+  final class Map4[K, +V](
+    private[collection] val key1: K,
+    private[collection] val value1: V,
+    private[collection] val key2: K,
+    private[collection] val value2: V,
+    private[collection] val key3: K,
+    private[collection] val value3: V,
+    private[collection] val key4: K,
+    private[collection] val value4: V) extends AbstractMap[K, V] with StrictOptimizedIterableOps[(K, V), Iterable, Map[K, V]] {
```

Comment:
It might be a little cleaner to add `private[collection] def buildTo(builder: HashMapBuilder)` and leave all the constructor vals private for consistency with `Map1-3`.

Reply:
👍